File: SafetyProgram.DocumentObjectUi/ChemicalTableNs/Ribbon/ChemicalTableRibbonViewModel.cs
using System;
using SafetyProgram.DocumentObjectUi.ChemicalTableNs.Commands;
using SafetyProgram.Models;

namespace SafetyProgram.DocumentObjectUi.ChemicalTableNs.Ribbon
{
    public sealed class ChemicalTableRibbonViewModel : IChemicalTableRibbonViewModel
    {
        public ChemicalTableRibbonViewModel(IChemicalTableCommands commands,
            IConfiguration configuration)
        { }

        public string Search
        {
            get
            {
                throw new NotImplementedException();
            }
            set
            {
                throw new NotImplementedException();
            }
        }

        public System.Collections.ObjectModel.ObservableCollection<Models.IChemical> SearchResult
        {
            get { throw new NotImplementedException(); }
        }

        public Commands.IChemicalTableCommands Commands
        {
            get { throw new NotImplementedException(); }
        }

        public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
    }
}

File: SafetyProgram.UI.ViewModels/ChemicalTableRibbonViewModel.fs
namespace SafetyProgram.UI.ViewModels

open SafetyProgram.UI.Views.ModelViews.ChemicalTableViews
open System.Collections.ObjectModel

type ChemicalTableRibbonViewModel() = 
    let propertyChanged = new Event<_,_>()
    let mutable search = ""
    let searchResult = new ObservableCollection<_>()

    interface IChemicalTableRibbonViewModel with
        member this.Search 
            with get () = search
            and set(x) = search<-x
        member this.SearchResult = searchResult
        [<CLIEvent>]
        member this.PropertyChanged = propertyChanged.Publish

